src/Pages/Signup/Signup.js

```diff
@@ -0,0 +1,347 @@
+import React, { Component } from 'react';
+import Header from '../../Component/HeaderComponent/Header';
+import Footer from '../../Component/FooterComponent/Footer';
+import DaumPostCode from 'react-daum-postcode';
+import { validateId } from '../../Utill/validaters';
+import { validateEmail } from '../../Utill/validaters';
+import { validatePhone } from '../../Utill/validaters';
+import './Signup.scss';
+import '../../Styles/Common.scss';
+
+// 주소창 style
+const width = 600;
+const height = 450;
+const postStyle = {
+  position: 'absolute',
+  zIndex: '100',
+  border: '1px solid #333333',
+};
+
+class Signup extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      identity: '',
+      background: '#d2f7d2',
+      password: '',
+      rePassword: '',
+      isAvailedPassword: '',
+      name: '',
+      phone: '',
+      email: '',
+      address: '',
+      firstAddress: '',
+      secondAddress: '',
+      isDaumPost: false,
+    };
+  }
+
+  //id 중복체크
+  checkDuplicateId = e => {
+    e.preventDefault();
+    fetch('http://10.58.2.234:8000/user/id-check', {
+      method: 'POST',
+      body: JSON.stringify({
+        identity: this.state.identity,
+      }),
+    })
+      .then(response => response.json())
+
+      .then(result => {
+        console.log('결과: ', result);
+        if (result.message === 'SUCCESS') {
+          alert('사용 가능한 아이디 입니다.');
+        } else if (result.message === 'IDENTITY_ALREADY_EXISTS') {
+          alert('이미 존재하는 아이디 입니다.');
+          this.setState({ identity: '' });
+        }
+      });
+  };
+
+  //pw 유효성검사 (알림창띄우기)
+  pwCheck = e => {
+    const pwreg =
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[$@$!%*?&])[A-Za-z\d$@$!%*?&]{8,18}/g;
+    let userPw = e.target.value;
+    this.setState({ password: e.target.value });
+
+    if (false === pwreg.test(userPw)) {
+      alert('다시 입력해주세요. (대소문자,숫자,특수문자 모두 포함 8~18자)');
+    }
```

Comment:
- let 말고 const 로 선언해주세요!
- 조건에서 비교대상을 왼쪽에 위치시켜주세요!
- `pwreg.test(userPw)` 자체가 boolean 타입이기 때문에 추가적으로 false 와 비교할 필요가 없습니다!
```suggestion
    const userPw = e.target.value;
    this.setState({ password: userPw });

    if (!pwreg.test(userPw)) {
      alert('다시 입력해주세요. (대소문자,숫자,특수문자 모두 포함 8~18자)');
    }
```